lua/tardis/cl_hud.lua

```diff
@@ -2,41 +2,73 @@
 
 surface.CreateFont("TARDIS-HUD-Large", {
     font="Roboto",
-    size=68
+    size=58
 })
 
 surface.CreateFont("TARDIS-HUD-Med", {
     font="Roboto",
-    size=40
+    size=36
 })
 
 surface.CreateFont("TARDIS-HUD-Small", {
     font="Roboto",
-    size=34
+    size=30
 })
 
-hook.Add("HUDPaint", "TARDIS-HUD", function()
-    if not TARDIS:GetSetting("health-enabled") then return end
-    if not (LocalPlayer():GetTardisData("interior") or LocalPlayer():GetTardisData("exterior")) then return end
-    local tardis = LocalPlayer():GetTardisData("exterior")
-    if not IsValid(tardis) then return end
-    local sw = ScrW()
-    local sh = ScrH()
-    local health = math.ceil(tardis:GetHealthPercent())
+local function CreatePercentageHUDPanel(text, value, offset, red_level)
+
+    red_level = red_level or 20
+
     local width = 115
-    if health >= 10 then width = width + 10 end
-    if health == 100 then width = width + 35 end
-    local height = (sw >= 800) and 120 or 95
-    local healthfont = (height == 120) and "TARDIS-HUD-Large" or "TARDIS-HUD-Med"
-    local x = (ScrW()-width)*0.02
-    local y = (ScrH()-height)*0.025
+    local height = (ScrW() >= 800) and 95 or 85
+
+    if value >= 10 then width = width + 10 end
+    if value == 100 then width = width + 20 end
+
+    local x = (ScrW() - width) * 0.02 + offset
+    local y = (ScrH() - height) * 0.025
+
+    local header_font = "TARDIS-HUD-Small"
+    local value_font = (height == 95) and "TARDIS-HUD-Large" or "TARDIS-HUD-Med"
+    local textcolor = (value > red_level) and NamedColor("FgColor") or NamedColor("Caution")
+
     draw.RoundedBox( 10, x, y, width, height, NamedColor("BgColor") )
-    local textcolor
-    local textcolor = (health > 0) and NamedColor("FgColor") or NamedColor("Caution")
-    if (health > 20) then textcolor = NamedColor("FgColor")
-    else textcolor = NamedColor("Caution") end
-    draw.DrawText( TARDIS:GetPhrase("Common.TARDIS"), "TARDIS-HUD-Small", x+10, y+10, textcolor, TEXT_ALIGN_LEFT )
-    draw.DrawText( tostring(health) .. "%", healthfont, x+10, y+45, textcolor, TEXT_ALIGN_LEFT )
+    draw.DrawText( text, header_font, x+10, y+10, textcolor, TEXT_ALIGN_LEFT )
+    draw.DrawText( tostring(value) .. "%", value_font, x+10, y+35, textcolor, TEXT_ALIGN_LEFT )
+
+    return width, height
+end
+
+hook.Add("HUDPaint", "TARDIS-HUD", function()
+    local ply = LocalPlayer()
+    if not (ply:GetTardisData("interior") or ply:GetTardisData("exterior")) then return end
```

Comment:
We can set the `tardis` variable up here to save us grabbing it twice and also we don't need to check the interior one at all, as that will break usage on no-interior scenarios

Reply:
befcee8d17bc313a91cad4e7940cbcbff17c1015